include/rocksdb/utilities/secondary_index.h

```diff
@@ -134,7 +134,7 @@ class SecondaryIndex {
   // For vector indices, the search target might be a vector, and the iterator
   // might return similar vectors from the index.
   virtual std::unique_ptr<Iterator> NewIterator(
-      const SecondaryIndexReadOptions& read_options,
+      const std::any& read_options,
```

Comment:
TIL! A quick question about the return type of this function, would it make sense to make it `SecondaryIndexIterator` and put the class's definition in this public header. Since only a subset of the `Iterator` APIs are supported, and its behavior is different from a regular iterator.